File: 프로그래머스/0/181902. 문자 개수 세기/문자 개수 세기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string my_string) {
    vector<int> answer(52, 0);

    for(const char& ch : my_string) {
        if(ch >= 'a') { // 소문자는 대문자 보다 뒤에 있음
            answer[ch - 'a' + 26]++;
        } else { // 대문자는 소문자보다 앞에 있음
            answer[ch - 'A']++;
        }
    }

    return answer;
}

File: 프로그래머스/0/181902. 문자 개수 세기/문자 개수 세기.swift
import Foundation

func solution(_ my_string:String) -> [Int] {
    
    var result: [Int] = Array(repeating: 0, count: 52)
    
    my_string.forEach {
        let ascii = Int(UnicodeScalar(String($0))!.value)
        if $0.isUppercase {
            result[ascii - 65] += 1
        } else {
            result[ascii - 71] += 1
        }
    }
    
    return result
}